src/temporal_logic/rules.rs

```diff
@@ -1119,65 +1134,64 @@ pub use combine_with_next_internal;
 
 /// Strengthen next with arbitrary number of predicates.
 /// pre:
-///     spec |= []next
 ///     spec |= []p1
 ///     spec |= []p2
 ///        ...
 ///     spec |= []pn
-///     stronger_next == next /\ p1 /\ p2 /\ ... /\ pn
+///     all == p1 /\ p2 /\ ... /\ pn
 /// post:
-///     spec |= []stronger_next
+///     spec |= []all
 ///
-/// Usage: strengthen_next_n!(stronger_next, spec, next, p1, p2, p3, p4)
+/// Usage: combine_spec_entails_always_n!(spec, all, p1, p2, p3, p4)
 #[macro_export]
-macro_rules! strengthen_next_n {
+macro_rules! combine_spec_entails_always_n {
     [$($tail:tt)*] => {
-        ::builtin_macros::verus_proof_macro_exprs!($crate::temporal_logic::rules::strengthen_next_n_internal!($($tail)*))
+        ::builtin_macros::verus_proof_macro_exprs!($crate::temporal_logic::rules::combine_spec_entails_always_n_internal!($($tail)*))
     }
 }
 
 #[macro_export]
-macro_rules! strengthen_next_n_internal {
-    ($stronger_next:expr, $spec:expr, $($tail:tt)*) => {
+macro_rules! combine_spec_entails_always_n_internal {
+    ($spec:expr, $all:expr, $($tail:tt)*) => {
         entails_always_and_n!($spec, $($tail)*);
-        temp_pred_equality(lift_action($stronger_next), combine_with_next!($($tail)*));
+        temp_pred_equality($all, combine_with_next!($($tail)*));
     };
 }
 
-pub use strengthen_next_n;
-pub use strengthen_next_n_internal;
+pub use combine_spec_entails_always_n;
+pub use combine_spec_entails_always_n_internal;
 
 /// Show that an spec entails the invariant by a group of action/state predicates which are also invariants entailed by spec.
 /// pre:
-///     spec |= []next
-///     forall |s, s_prime| next(s, s_prime) ==> inv(s, s_prime)
+///     spec |= []partial_spec
+///     partial_spec |= inv
 ///     spec |= []p1
 ///     spec |= []p2
 ///         ...
 ///     spec |= []pn
-///     next == p1 /\ p2 /\ ... /\ pn
+///     partial_spec == p1 /\ p2 /\ ... /\ pn
 /// post:
 ///     spec |= []inv
-/// 
-/// Usage: invariant_action_n!(spec, next, inv, p1, p2, ..., pn)
+///
+/// Usage: invariant_n!(spec, partial_spec, inv, p1, p2, ..., pn)
 #[macro_export]
-macro_rules! invariant_action_n {
+macro_rules! invariant_n {
```

Comment:
Not sure if I missed sth. Could you point me to where you used this lemma?

Reply:
In `invariants.rs`.